### DISCUSSION TITLE
为什么像`@modern-js/module-tools`这样的工具没有单独仓库？

### DISCUSSION THREAD

**Discussion by zhangfisher** [web-infra-dev/modern.js#3580] [Q&A] [answered]
### What problem does this feature solve?

为什么像`@modern-js/module-tools`这样的工具没有单独仓库？

全部搞在一起，都不好看更新历史

### What does the proposed API look like?

如上题

**Comment by chenjiahan** [comment, marked as answer]
因为 Modern.js 里的各个解决方案并不是完全独立的，它们共享了很多核心包，比如 `@modern-js/core` `@modern-js/utils` `@modern-js/plugin` 等等。这些包提供了一致的插件机制、工具方法和通用三方依赖。

在同一个 monorepo 中，我们可以很方便地修改底层包，并测试它的改动对各个解决方案的影响。

同时，我们也可以保证这些 packages 的版本是通过 workspace 协议统一管理和升级的，避免出现版本不一致的问题。

如果要看更新历史，可以点开各个 package 的 CHANGELOG.md 文件来查看各个包的改动历史，比如 https://github.com/web-infra-dev/modern.js/blob/main/packages/solutions/module-tools/CHANGELOG.md。

后续我们会考虑为各个解决方案提供一份整理后的 Changelog，来更为清晰地展示改动内容。
